compiler/passes/ResolveScope.cpp

```diff
@@ -437,6 +454,58 @@ bool ResolveScope::extend(VisibilityStmt* stmt) {
   return true;
 }
 
+// If the new function is a method, track the type on which it is defined, if
+// we can.
+void ResolveScope::extendMethodTracking(FnSymbol* newFn) {
+  if (newFn != NULL) {
+    if (newFn->_this != NULL && newFn->hasFlag(FLAG_PRIVATE) == false) {
+      ArgSymbol* _this = toArgSymbol(newFn->_this);
+      INT_ASSERT(_this);
+      // New function is a method.  The type on which this method is defined
+      // might be listed in a limitation clause on a use or import statement, so
+      // we should track the type on which it is defined, if we can and it isn't
+      // private
+      if (_this->typeInfo() != dtUnknown) {
+        // The type is already known, so just use that information
+        mMethodsOnTypeName.insert(astr(_this->typeInfo()->symbol->name));
```

Comment:
no need for astr call - name should already be an astr. If you are worried about it you can add a check under an `fVerify` condition.

Reply:
This applies also to the other uses of `astr` in this function